CMakeLists.txt

```diff
@@ -1001,6 +1001,8 @@ include(${PROJECT_SOURCE_DIR}/vendor/unique_resource.cmake)
 include(${PROJECT_SOURCE_DIR}/vendor/vector-tile.cmake)
 include(${PROJECT_SOURCE_DIR}/vendor/wagyu.cmake)
 
+find_package(Threads REQUIRED)
```

Comment:
This should be done conditionally when and where it is actually used.

Reply:
Moved it to the linux platform's CMake

Reply:
Thanks! It looks better, but I still wonder if all configurations on Linux need it. The reason I'm asking is because we build MapLibre in so many configurations and we do not want to pull in any unwanted dependencies.

Tagging also @acalcutt if he ever stumbled upon this issue with Node.js.

Also @louwers, we have another CI not running case here.

Reply:
Which version of debian is this being built on? 

I have run into an issue with pthread in the past, but I haven't had an issues with it in a while. Looking back it may have been when I tested building the QT version in https://github.com/maplibre/maplibre-gl-native/issues/280#issuecomment-1123907951

Reply:
Can you list the OS and build commands you are using?

Reply:
My server is debian 11 bullseye, kernel 5.10.0-20-amd64. A standard configure & build is enough to replicate this on my end.

Documented configure & build with gcc10 + ninja:
```
cmake . -B build -G Ninja -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10
cmake --build build
```
And a simple configure & build with clang:
```
cmake . -B build -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++
cmake --build build
```

Reply:
@ntadej There is no CI for the linux platform.

Reply:
Created an issue: https://github.com/maplibre/maplibre-gl-native/issues/967